feat: add a post creation form

mwawrzen opened the pull request.

mwawrzen committed:
feat: create post add form

mwawrzen committed:
feat: add collection price form transition

mwawrzen committed:
feat: add image form

mwawrzen committed:
feat: add url input

Wuzado committed:
fix: change the path of the post creation form

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+              label="zdjęcia"
+              max="2"
+            ></v-file-input>
+            <v-btn class="ml-8 mt-5" style="color: #eee; background: #5570d5">
+              dodaj
+            </v-btn>
```
This button does nothing.

Wuzado replied:
Duplicate comment, closing.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+              small-chips
+              multiple
+              label="zdjęcia"
+              max="2"
```
It doesn't seem that there is a limit for the images added.

Wuzado replied:
If that's a limit on the amount of files sent, it seems to not work. Also, 2 images seem like a pretty low number. That's something to discuss, but I guess 5 (or even 10?) would be better?

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+            ></v-file-input>
+            <v-btn class="ml-8 mt-5" style="color: #eee; background: #5570d5">
+              dodaj
+            </v-btn>
```
As of now, this button does nothing.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+            dense
+            required
+          ></v-text-field>
+        </v-expand-transition>
```
I'm not sure that's something we want to implement in the end: I think for that we'd also need a current value of the post, and it'll be hard to synchronize that with the actual sites, primarily because of private APIs/TOS breaking site scrapping. I assume people can simply check the site of the fundraising campaign itself...

It's definitely something to look into, though: perhaps talking with some existing fundraising sites might help?

Wuzado replied:
Temporarily removed in a54dfa2.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+        (v) => !!v || 'Uzupełnij puste pola',
+        (v) => v.length <= 100 || 'Tytuł nie spełnia wymagań',
+      ],
+      descRules: [(v) => !!v || 'Uzupełnij puste pola'],
```
Add the 2000 character limit from the backend. This would also require adding a new counter for the description field.

Wuzado replied:
Added in b8b313d.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+      url: '',
+      title: '',
+      desc: '',
+      files: [],
```
I assume this is dependent on the implementation on the backend, but I think this should be wrapped in an object, which is POSTed to the backend.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+              multiple
+              label="zdjęcia"
+              max="2"
+            ></v-file-input>
```
You can add multiple files only in the same dialogue: clicking the field again makes the files reset. You should also be able to discard the files separately, and not necessarily only as a whole. I assume this would most likely require us to build a more advanced mechanism for file input. That field also could use some kind of a preview, too.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+          <v-textarea
+            v-model="desc"
+            label="opis"
+            height="300"
```
I honestly don't know if there's need for set height/no-resize... The default values look decent, and this gives the user more comfort, and is a little QoL feature.

Wuzado replied:
Tweaked in 8104e02.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+            label="link do zbiórki"
+            outlined
+            dense
+          ></v-text-field>
```
I think this should be reworked, either by changing the wording to just include any pages about the campaign (no matter what campaign it is), or be only for fundraising campaigns.

Personally, I prefer the first option.

Wuzado replied:
Tweaked in fa25725.

Wuzado commented on pages/add-post.vue:
```diff
@@ -0,0 +1,133 @@
+</script>
+
+<style scoped>
+.v-form,
```
Not sure if that has any effect on v-form.

Wuzado replied:
Removed in e972b79.

Wuzado reviewed (changes requested):
More general thing: capitalize first letters of the placeholders.

EDIT: also, change the URL to make it uniform with mock-ui (#8) branch.

Wuzado committed:
fix(post-add): remove the hard boundaries for the description field

Wuzado committed:
fix(post-add): make the post form fit a bit better on mobile devices

Wuzado committed:
fix(post-add): temporarily remove the amount of fundraising money

Reasoning in the respective PR, TL;DR: we need to actually cooperate with fundraising sites.

Wuzado committed:
feat(post-add): add the description limit

Wuzado committed:
fix(post-add): change the wording of the post link in the add form

Wuzado committed:
fix(post-add): capitalize field labels

Wuzado committed:
fix(post-add): remove the redundant style

Wuzado committed:
fix(post-add): change the wording of the title error message

Wuzado committed:
feat(post-add): add the URL limit
